projects/demo/src/app/app.component.ts

```diff
@@ -156,8 +162,9 @@ export class AppComponent implements OnInit, OnDestroy {
   }
 
   public ngOnInit(): void {
-    this._subscription = this.darkMode.valueChanges.subscribe((checked) => {
-      this._changeTheme(checked ? 'dark' : 'light');
+    this._subscription = this.darkMode.valueChanges.subscribe((value) => {
+      const isDark = value === 'dark-mode';
+      this._changeTheme(isDark ? 'dark' : 'light');
```

Comment:
if the theme values would be dark or light instead of dark-mode and light-mode, then the value comparison (`value === 'dark-mode'`) could be skipped